woof-code/rootfs-skeleton/usr/sbin/micro2pup.sh

```diff
@@ -0,0 +1,252 @@
+#!/bin/sh
+
+# GUI to download early or late loading microcode for AMD and Intel
+# GPLv2
+
+. /etc/rc.d/PUPSTATE
+. /etc/DISTRO_SPECS
+
+
+exit_error() {
+	box_ok "Error" error "$1"
+	exit 1
+}
+
+info_msg() {
+	box_ok "Info" info "$1"
+}
+EARLY=0 # true
+
+#### exclusions and special cases
+MSG1=$(gettext 'Your initrd.gz resides on an unsupported file system, however late loading may work if your main filesystem is writeable.')
+MSG2=$(gettext 'Unsupported CPU Architecture.')
+MSG3=$(gettext 'This program does not support a Virtual Machine.')
+MSG4=$(gettext 'The puppy home drive is read only, however if saving session late loading is supported.')
+MSG5=$(gettext 'Your CPU vendor is unsupported.')
+PUPDIR=/initrd${PUP_HOME}${PUPSFS##*,}
```

Comment:
What about 'full' installations? The fixes I'll need to do for `frugalify`/PUPMODE=6 images built in https://github.com/dpupos/woof-CE are the same or almost the same as those needed to support early microcode loading in 'full' installations.

Reply:
Full installs put `vmlinuz` in `/boot` IIRC, so if `ucode.cpio` is there and bootloader calls it shouldn't be an issue.